shopinvader_delivery_carrier/services/delivery.py

```diff
@@ -0,0 +1,117 @@
+# -*- coding: utf-8 -*-
+# Copyright 2019 ACSONE SA/NV (<http://acsone.eu>)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+from odoo import fields
+from odoo.addons.base_rest.components.service import to_int
+from odoo.addons.component.core import Component
+
+
+class AbstractSaleService(Component):
+    _inherit = "base.shopinvader.service"
+    _name = "shopinvader.delivery.service"
+    _usage = "delivery"
+    _expose_model = "stock.picking"
+
+    def search(self, **params):
+        """
+        Get every delivery related to logged user
+        :param params: dict/json
+        :return: dict
+        """
+        return self._paginate_search(**params)
+
+    def _validator_search(self):
+        """
+        Validator for the search
+        :return: dict
+        """
+        schema = {
+            "per_page": {
+                "coerce": to_int,
+                "nullable": True,
+                "type": "integer",
+            },
+            "page": {"coerce": to_int, "nullable": True, "type": "integer"},
+        }
+        return schema
+
+    def _get_parser_picking(self):
```

Comment:
@acsonefho Why not an ir.export instance? If we start to 'hard code' the returned result it becomes impossible to add a new field into the result without modifying the code....

Reply:
@lmignon I don't understand why I should add this? Into shopinvader modules, I don't see any module who is using an `ir.exports` instance as a parser.
And to have the related `ir.exports`, I should search on the `resource = 'stock.picking'` but if another module/service add a new exporter (who doesn't match with mine), how I can manage it?
Thanks